packages/next/shared/lib/dynamic.tsx

```diff
@@ -65,29 +65,33 @@ export default function dynamic<P = {}>(
   options?: DynamicOptions<P>
 ): React.ComponentType<P> {
   let loadableFn: LoadableFn<P> = Loadable
-  let loadableOptions: LoadableOptions<P> = {
-    // A loading component is not required, so we default it
-    loading: ({ error, isLoading, pastDelay }) => {
-      if (!pastDelay) return null
-      if (process.env.NODE_ENV === 'development') {
-        if (isLoading) {
+
+  let loadableOptions: LoadableOptions<P> = options?.suspense
```

Comment:
We can delete all ignorable options after the checking in dev mode

```
if (suspense) {
  if (dev) {
     // warning for the options will be ignored
  }
  delete options.loading
  delete options.ssr
}
```

Reply:
Sure!

Reply:
I think we can revert the change here then

Reply:
Probably not.

We have to distinguish if the `loading` is either from Next.js' default or is user-provided, and only print a warning if `loading` is provided by the user.